src/gt4py/backend/base.py

```diff
@@ -153,7 +153,7 @@ def extra_cache_validation_keys(self) -> List[str]:
         return []
 
 
-class CLIBackendMixin:
+class CLIBackendMixin(Backend):
```

Comment:
This is the recommended way for mypy to recognize that something which is of type `CLIBackendMixin` has to also be a `Backend` and that it is allowed to call the methods of a `Backend` on it. Other ways of achieving this would require building a separate class hierarchy for the benefit of mypy or rearranging the current one (which is out of scope here).